File: leetcode/1201-1500/1351-1400/1365.How Many Numbers Are Smaller Than the Current Number.cc
// https://leetcode.com/problems/how-many-numbers-are-smaller-than-the-current-number/

class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> smaller_count(nums.size());
        for (unsigned i = 0; i < nums.size(); ++i) {
            int& cur_num = nums[i];
            smaller_count[i] = count_if(nums.begin(), nums.end(), [&cur_num](int& num){
                return cur_num > num;
            });
        }
        return move(smaller_count);
    }
};

File: leetcode/1201-1500/1351-1400/1365.How Many Numbers Are Smaller Than the Current Number.rs
// https://leetcode.com/problems/how-many-numbers-are-smaller-than-the-current-number

impl Solution {
    pub fn smaller_numbers_than_current(nums: Vec<i32>) -> Vec<i32> {
        nums.iter().map(|n| {
            nums.iter().filter(|&other| *other < *n).count() as i32
        }).collect::<Vec<i32>>()
    }
}